sdk/eventhub/Azure.Messaging.EventHubs/src/Compatibility/EventHubClientBuilderExtensions.cs

```diff
@@ -46,6 +47,8 @@ public static IAzureClientBuilder<EventHubProducerClient, EventHubProducerClient
         ///   Registers a <see cref="EventHubProducerClient"/> instance with connection options loaded from the provided <paramref name="configuration"/> instance.
         /// </summary>
         ///
+        [RequiresUnreferencedCode("Binding strongly typed objects to configuration values requires generating dynamic code at runtime, for example instantiating generic types. Use the Configuration Binder Source Generator (EnableConfigurationBindingGenerator=true) instead.")]
```

Comment:
> requires generating dynamic code at runtime

(nit) This RequiresUnreferencedCode attribute has a message saying it requires generating dynamic code.